documentation/book/assembly-upgrade.adoc

```diff
@@ -5,18 +5,81 @@
 [id='assembly-upgrade-{context}']
```

Comment:
This assembly has a lot of content in it - it should be mostly includes to modules.  There are any concepts and reference material here that should all be split out.

Reply:
Agreed. I will do some work to reduce the content down. It might be a separate issue so I can concentrate on improving the content here.